### PR TITLE
Submissions list improvements

### PR DESCRIPTION
**Pull request by ckcollab**
# A brief description of the purpose of the changes contained in this PR.
Improves submission table (show's participant submission numbers, is searchable, loads all submissions at once)

### CODE REVIEW

**Comment by ckcollab** [comment]
@Tthomas63 can you please take a look at this? we'll be merging this in before py 2->3 stuff as well, should be _mostly_ ready to go, probably a few places I can improve

File: codalab/apps/api/routers.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 router.register(r'competition', views.CompetitionAPIViewSet)
 router.register(r'competition/(?P<competition_id>\d+)/leaderboards', views.LeaderBoardViewSet)
 router.register(r'defaultcontent', views.DefaultContentViewSet)
+# router.register(r'submissions', views.CompetitionSubmissionListViewSet, )
```
Comment (ckcollab) [inline codalab/apps/api/routers.py:15]: will pull cruft

File: codalab/apps/api/views/competition_views.py [hunk 2]
```diff
@@ -646,11 +645,92 @@ def addToLeaderboard(self, request, pk=None, competition_id=None):
...
+            Competition.objects.get(pk=competition_id)
+        )
+
+        # Get all submissions that can be migrated into another phase (must be on leaderboard already and
+        # the next phase must have auto_migration
+        if next_phase and next_phase.auto_migration:
+            active_phase_submissions = CompetitionSubmission.objects.filter(
+                phase=active_phase,
+            ).values_list('id', flat=True)
+            context['migratable_submissions'] = filter(lambda x: x in context["leaderboard_submissions"], active_phase_submissions)
```
Comment (Tthomas63) [inline codalab/apps/api/views/competition_views.py:724]: Would a list comprehension work here?

Py3 seems to prefer list comprehension over `filter(lambda)`

IE:
```
context['migratable_submissions'] = [submission for submission in context["leaderboard_submissions"] if submission in active_phase_submissions]
```

It looks like list comprehensions might be faster than filter(lambda) as well.

File: codalab/apps/api/views/competition_views.py [hunk 3]
```diff
@@ -646,11 +645,92 @@ def addToLeaderboard(self, request, pk=None, competition_id=None):
         response['status'] = (201 if cr else 200)
         return Response(response, status=response['status'], content_type="application/json")
 
+    def destroy(self, request, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (ckcollab) [reply codalab/apps/api/views/competition_views.py:648]: I think that's cruft? Does it even get called?

[the remark this answers is not in the document]
Comment (Tthomas63) [reply codalab/apps/api/views/competition_views.py:648]: It's not commented out, but it also doesn't have an `action` decorator either, so you're probably right, I'm not sure if you could actually hit it at the moment.

**Comment by ckcollab** [comment]
@Tthomas63 did you hand test this at all? any chance you can review the vue bits even a cursory glance? still a smidge of work to do in supporting some pieces.. (i.e. migrating submissions between phases)

**Comment by Tthomas63** [comment]
I did check it out locally, but I did not hand-test extensively due to not having a storage bucket that will work with `path` style addressing. I'll send you an email in regards to this.

**Comment by ckcollab** [comment]
As per email, you should be able to setup minio?

**Comment by Tthomas63** [comment]
It looks like we might be able to reduce the width of some of the columns (ID's) and eliminate the need for horizontal scrolling. 

image

**Comment by ckcollab** [comment]
@Tthomas63 ready for another pass, then we can merge it into py3 + that staging server for further testing

**Comment by Tthomas63** [comment]
Migration button seems to work just fine. The one thing I did miss previously was the mark submission as failed button throws an error for me:
```
vue.js:634 [Vue warn]: Property or method "fail_submission" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.

(found in <Root>)
warn @ vue.js:634
vue.js:634 [Vue warn]: Error in v-on handler: "TypeError: fail_submission is not a function"

found in

---> <VBtn>
       <VTooltip>
         <VBtnToggle>
           <VSimpleTable>
             <VData>
               <VDataTable>
                 <VCard>
                   <VApp>
                     <Root>
warn @ vue.js:634
vue.js:1897 TypeError: fail_submission is not a function
    at click (eval at createFunction (vue.js:1), <anonymous>:3:5683)
    at invokeWithErrorHandling (vue.js:1863)
    at VueComponent.invoker (vue.js:2188)
    at invokeWithErrorHandling (vue.js:1863)
    at VueComponent.Vue.$emit (vue.js:3889)
    at VueComponent.click (VBtn.ts:157)
    at invokeWithErrorHandling (vue.js:1863)
    at HTMLButtonElement.invoker (vue.js:2188)
    at HTMLButtonElement.original._wrapper (vue.js:7547)
```

The errors only show in console once the button has been clicked (not at page load). I tried collecting static + refreshing without cache and the issue still persists.

**Comment by ckcollab** [comment]
try out latest push?

**Comment by Tthomas63** [comment]
Works now! 👍

**Pull request merged**